src/GraphQLClient.js

```diff
@@ -0,0 +1,131 @@
+class GraphQLClient {
+  constructor(config) {
+    this.cache = config.cache;
+    this.headers = config.headers || [];
```

Comment:
Should this be an object? We're treating it as one in the rest of the file
```suggestion
    this.headers = config.headers || {};
```